plugins/inputs/fibaro/fibaro.go

```diff
@@ -0,0 +1,183 @@
+package fibaro
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/influxdata/telegraf/plugins/inputs/httpjson"
+)
+
+const sampleConfig = `
+  ## Required Fibaro controller address/hostname.
+  ## Note: connection is done over http/80 as Fibaro did not implement https.
+  server = "<controller>"
+  ## Required credentials to access the API (http://<controller/api/<component>)
+  username = "<username>"
+  password = "<password>"
+`
+
+const description = "Read devices value(s) from a Fibaro controller"
+
+// Fibaro contains connection information
+type Fibaro struct {
+	Server   string
+	Username string
+	Password string
+
+	client httpjson.HTTPClient
+}
+
+// LinkRoomsSections links rooms to sections
+type LinkRoomsSections struct {
+	Name      string
+	SectionID uint16
+}
+
+// Sections contains sections informations
+type Sections struct {
+	ID   uint16 `json:"id"`
+	Name string `json:"name"`
+}
+
+// Rooms contains rooms informations
+type Rooms struct {
+	ID        uint16 `json:"id"`
+	Name      string `json:"name"`
+	SectionID uint16 `json:"sectionID"`
+}
+
+// Devices contains devices informations
+type Devices struct {
+	ID         uint16 `json:"id"`
+	Name       string `json:"name"`
+	RoomID     uint16 `json:"roomID"`
+	Type       string `json:"type"`
+	Enabled    bool   `json:"enabled"`
+	Properties struct {
+		Dead   interface{} `json:"dead"`
+		Value  interface{} `json:"value"`
+		Value2 interface{} `json:"value2"`
+	} `json:"properties"`
+}
+
+// Description returns a string explaining the purpose of this plugin
+func (f *Fibaro) Description() string { return description }
+
+// SampleConfig returns text explaining how plugin should be configured
+func (f *Fibaro) SampleConfig() string { return sampleConfig }
+
+// getJSON connects, authenticates and reads JSON payload returned by Fibaro box
+func (f *Fibaro) getJSON(path string, dataStruct interface{}) error {
+	var requestURL = "http://" + f.Server + path
+
+	req, err := http.NewRequest("GET", requestURL, nil)
+	if err != nil {
+		return err
+	}
+
+	req.SetBasicAuth(f.Username, f.Password)
+	resp, err := f.client.MakeRequest(req)
+	if err != nil {
+		return err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		err = fmt.Errorf("Response from url \"%s\" has status code %d (%s), expected %d (%s)",
+			requestURL,
+			resp.StatusCode,
+			http.StatusText(resp.StatusCode),
+			http.StatusOK,
+			http.StatusText(http.StatusOK))
+		return err
+	}
+
+	defer resp.Body.Close()
+
+	dec := json.NewDecoder(resp.Body)
+	err = dec.Decode(&dataStruct)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Gather fetches all required information to output metrics
+func (f *Fibaro) Gather(acc telegraf.Accumulator) error {
+
+	if f.client.HTTPClient() == nil {
+		f.client.SetHTTPClient(&http.Client{})
```

Comment:
Follow the style of the http input to create the client, this way the plugin has its own transport and timeout, and will use the system proxy settings.

https://github.com/influxdata/telegraf/blob/master/plugins/inputs/http/http.go#L99-L112